docs/security-best-practices.md

```diff
@@ -153,6 +153,87 @@ To change the default endpoint to be `localhost`-bound in all components, enable
 
 If `localhost` resolves to a different IP due to your DNS settings then explicitly use the loopback IP instead: `127.0.0.1` for IPv4 or `::1` for IPv6. In IPv6 setups, ensure your system supports both IPv4 and IPv6 loopback addresses to avoid issues.
 
+Using `localhost` may not work in environments like Docker, Kubernetes, and other environments that have non-standard networking setups.  We've documented a few working example setups for the OTLP receiver gRPC endpoint below, but other receivers and other Collector components may need similar configuration.
+
+
+#### Docker
+```yaml
+services:
+  otel-collector:
+    image: otel/opentelemetry-collector-contrib:0.104.0
+    ports:
+      - "4567:4317"
+```
+
+otel collector config file:
+```yaml
+receivers:
+  otlp:
+    protocols:
+      grpc:
+         endpoint: otel-collector:4317 # Using the service name from your Docker compose file
+```
+
+You can connect to this collector from another Docker container running in the same network as the Collector by connecting to `otel-collector01:4317`.  You could access it from outside that Docker network (for example on a regular program running on the host) by connecting to `127.0.0.1:4567`.
+
+#### Kubernetes
+If you run the collector as a `Daemonset`, you can use a configuration like below:
+```yaml
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: collector
+spec:
+  selector:
+    matchLabels:
+      name: collector
+  template:
+    metadata:
+      labels:
+        name: collector
+    spec:
+      containers:
+        - name: collector
+          image: otel/opentelemetry-collector:0.104.0
+          volumeMounts:
+            - mountPath: /etc/otelcol/
+              name: config
+          ports:
+            - containerPort: 4317
+              hostPort: 4317
+              protocol: TCP
+              name: otlp
+          env:
+            - name: MY_POD_IP
+              valueFrom:
+                fieldRef:
+                  fieldPath: status.podIP
+      volumes:
+        - name: config
+          configMap:
+            name: collector-config
+
+```
+In this example, we use the [Kubernetes Downward API](https://kubernetes.io/docs/concepts/workloads/pods/downward-api/) to get your own Pod IP, then bind to that network interface.  Then, we use the hostPort option to ensure that the Collector is exposed on the host.  The Collector's config should look like:
+
+```yaml
+receivers:
+  otlp:
+    protocols:
+      grpc:
```

Comment:
Should we include an `http` endpoint additionally? Check the Helm Chart defaults: https://github.com/open-telemetry/opentelemetry-helm-charts/blob/5e5e4354263055e19de5275b13939c1addd52d58/charts/opentelemetry-collector/values.yaml#L113-L118

Reply:
sure!